src/Symfony/Component/PropertyInfo/Type.php

```diff
@@ -77,8 +85,10 @@ public function __construct(string $builtinType, bool $nullable = false, string
      * Gets built-in type.
      *
      * Can be bool, int, float, string, array, object, resource, null, callback or iterable.
+     *
+     * @return string|array
      */
-    public function getBuiltinType(): string
+    public function getBuiltinType()
```

Comment:
Code that interacts with `PropertyInfo` might not expect to receive an array from `getBuiltinType()`. This is a breaking change imho.

Reply:
Maybe not: BC layers are for already written code. No existing code should return an array here, so we might not break any. For thoughts.

Reply:
this is indeed forbidden by the BC policy: https://symfony.com/doc/current/contributing/code/bc.html#changing-classes (scroll to see the bottom of the `Public methods` section of the table)

Reply:
@nicolas-grekas Already written code calling `getBuiltinType()` will expect to receive a string. That code will break.